iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -114,32 +137,69 @@ TEST_F(ServiceDescription_test, CtorNoParams)
 
 TEST_F(ServiceDescription_test, CtorIDStrings)
 {
-    IdString_t testService("Service");
-    IdString_t testInstance("Instance");
-    IdString_t testEvent("Event");
+    IdString_t testService1("Service");
+    IdString_t testInstance1("Instance");
+    IdString_t testEvent1("Event");
+    IdString_t testService2("65536");
+    IdString_t testInstance2("65536");
+    IdString_t testEvent2("65536");
     ServiceDescription::ClassHash testHash = {1, 2, 3, 4};
-    ServiceDescription csd = ServiceDescription(testService, testInstance, testEvent, testHash);
-
-    EXPECT_THAT(csd.getServiceIDString(), StrEq("Service"));
-    EXPECT_THAT(csd.getInstanceIDString(), StrEq("Instance"));
-    EXPECT_THAT(csd.getEventIDString(), StrEq("Event"));
-    EXPECT_EQ(uint32_t(1), csd.getClassHash()[0]);
-    EXPECT_EQ(uint32_t(2), csd.getClassHash()[1]);
-    EXPECT_EQ(uint32_t(3), csd.getClassHash()[2]);
-    EXPECT_EQ(uint32_t(4), csd.getClassHash()[3]);
+    ServiceDescription csd1 = ServiceDescription(testService1, testInstance1, testEvent1, testHash);
+    ServiceDescription csd2 = ServiceDescription(testService2, testInstance2, testEvent2, testHash);
+    EXPECT_THAT(csd1.getServiceIDString(), StrEq("Service"));
```

Comment:
Code duplication. Since you are here testing somehow the same thing twice for different inputs you could create either a lambda when you only need this in this test case or create a free function. 
This would improve the readability very much.

Please look also for code duplication in other tests and abstract them to improve the code readability.